functionaltests/steps.go

```diff
@@ -314,8 +337,97 @@ type customStep struct {
 	Func stepFunc
 }
 
-var _ testStep = customStep{}
-
 func (c customStep) Step(t *testing.T, ctx context.Context, e *testStepEnv, previousRes testStepResult) testStepResult {
 	return c.Func(t, ctx, e, previousRes)
 }
+
+// expectedDataStreamsIngest represent the expected number of ingested document
+// after a single run of ingest.
+//
+// NOTE: The aggregation data streams have negative counts, because they are
+// expected to appear but the document counts should not be asserted.
+func expectedDataStreamsIngest(namespace string) esclient.DataStreamsDocCount {
+	return map[string]int{
+		fmt.Sprintf("traces-apm-%s", namespace):                     15013,
+		fmt.Sprintf("metrics-apm.app.opbeans_python-%s", namespace): 1437,
+		fmt.Sprintf("metrics-apm.internal-%s", namespace):           1351,
+		fmt.Sprintf("logs-apm.error-%s", namespace):                 364,
+		// Ignore aggregation data streams.
+		fmt.Sprintf("metrics-apm.service_destination.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_transaction.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_summary.1m-%s", namespace):     -1,
+		fmt.Sprintf("metrics-apm.transaction.1m-%s", namespace):         -1,
+	}
+}
+
+// emptyDataStreamsIngest represent an empty ingestion.
+// It is useful for asserting that the document count did not change after an operation.
+//
+// NOTE: The aggregation data streams have negative counts, because they
+// are expected to appear but the document counts should not be asserted.
+func emptyDataStreamsIngest(namespace string) esclient.DataStreamsDocCount {
+	return map[string]int{
+		fmt.Sprintf("traces-apm-%s", namespace):                     0,
+		fmt.Sprintf("metrics-apm.app.opbeans_python-%s", namespace): 0,
+		fmt.Sprintf("metrics-apm.internal-%s", namespace):           0,
+		fmt.Sprintf("logs-apm.error-%s", namespace):                 0,
+		// Ignore aggregation data streams.
+		fmt.Sprintf("metrics-apm.service_destination.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_transaction.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_summary.1m-%s", namespace):     -1,
+		fmt.Sprintf("metrics-apm.transaction.1m-%s", namespace):         -1,
+	}
+}
```

Comment:
Why are aggregated metrics data streams not expected to be 0 if no docs were indexed?

Reply:
This ties to the comment [above](https://github.com/elastic/apm-server/pull/16917/files#r2101712325). The reason why the are ignored is because the aggregation doc count can change independently from ingestion. This leads to tests failing based on timing.

The way this variable is used, similarly to `expectedDataStreamsIngest`, is to add a certain amount to the current known amount and use this value in assertions. As the doc count for aggregation data streams is not known, they are ignored in this case too.

But if we move in a different direction, where we get the document count immediately before the action that (may) affect it, we can also get rid of the logic that uses these functions. I think it's a simplification worth pursuing.